cmd/monaco/v2/deploy/deploy.go

```diff
@@ -399,18 +418,21 @@ func getApiNames(apis map[string]api.Api) []string {
 	return result
 }
 
-func filterEnvironmentByName(environments []manifest.EnvironmentDefinition, name string) ([]manifest.EnvironmentDefinition, error) {
+func filterEnvironmentsByName(environments []manifest.EnvironmentDefinition, names []string) ([]manifest.EnvironmentDefinition, error) {
 	var result []manifest.EnvironmentDefinition
 
-	for _, env := range environments {
-		if env.Name == name {
-			result = append(result, env)
+	for _, name := range names {
+		found := false
+		for _, env := range environments {
+			if env.Name == name {
+				result = append(result, env)
+				found = true
+			}
+		}
+		if !found {
+			return nil, fmt.Errorf("no environment found `%s`", name)
```

Comment:
Could we maybe collect all potential errors here?
I guess it is better for the end user if they see in one
go, what envs do not exist.

Reply:
We might want to do the same for projects

Reply:
It's already done for projects 
https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/blob/af39771231aedecad2a4558da557123d95c2d7d4/cmd/monaco/v2/deploy/deploy.go#L323